### PR TITLE
Fix links in From, Into, Cursor docs

### PR DESCRIPTION
**Pull request by ghost**
Tidied up some more issues with quote/bracket nesting. These are the last ones that I could easily find.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/convert.rs [hunk 1]
```diff
@@ -178,14 +178,14 @@ pub trait Into<T>: Sized {
 /// ```
 /// # Generic impls
 ///
-/// - `From<T> for U` implies `[Into<U>] for T`
+/// - `From<T> for U` implies [`Into<U>`]` for T`
```
Comment (GuillaumeGomez) [inline src/libcore/convert.rs]: Not sure to understand why this change is useful...

**Comment by GuillaumeGomez** [comment]
Just a point which needs explanation otherwise it's all good.

File: src/libcore/convert.rs (hunk 1 above)
Comment (ollie27) [inline src/libcore/convert.rs]: You can see how this renders [here](https://doc.rust-lang.org/nightly/std/convert/trait.From.html). Basically you can't have links inside code blocks.

Before:
 - `From<T> for U` implies `[Into<U>] for T`
[Into<U>]: trait.Into.html

After:
 - `From<T> for U` implies [`Into<U>`]` for T`
[`Into<U>`]: trait.Into.html

**Comment by ollie27** [comment]
I've also fixed these as part of #37316.

**Comment by vtduncan** [comment]
@ollie27 Thanks and well spotted! This pull is a dupe then.
